File: Latihan Kelas/KendaraanInheritance/Kendaraan.cpp
#include "Kendaraan.hpp"
#include <iostream>
using namespace std;

//default constructor
Kendaraan::Kendaraan(){ 
    this->nomor = 0;
    this->tahun = 0;
    this->merk = "XXX";
}

//user defined constructor
Kendaraan::Kendaraan(int nomor, int tahun, string merk){
    this->nomor = nomor;
    this->tahun = tahun;
    this->merk = merk;
}

//mencetak semua informasi yang dimiliki oleh kendaraan
void Kendaraan::printInfo(){
    cout << "Nomor: " << this->nomor << endl;
    cout << "Tahun: " << this->tahun << endl;
    cout << "Merk: " << this->merk << endl;
}

File: Latihan Kelas/KendaraanInheritance - Jawwa/Kendaraan.java
public class Kendaraan{
    protected int nomor;
    protected int tahun;
    protected String merk;

    public Kendaraan(){
        this.nomor = 0;
        this.tahun = 0;
        this.merk = "XXX";
    }

    public Kendaraan(int nomor, int tahun, String merk){
        this.nomor = nomor;
        this.tahun = tahun;
        this.merk = merk;
    }

    public Kendaraan(Kendaraan k){
        this.nomor = k.nomor;
        this.tahun = k.tahun;
        this.merk = k.merk;
    }

    public void printInfo(){
        System.out.println("Nomor: " + nomor);
        System.out.println("Tahun: " + tahun);
        System.out.println("Merk: " + merk);
    }
}